shopinvader_delivery_pickup_public_edition/services/cart.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo.addons.component.core import Component
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+
+class CartService(Component):
+    _inherit = "shopinvader.cart.service"
+
+    def set_public_delivery_pickup(self, **params):
```

Comment:
@sebastienbeau IMO we should define a method into the pickup service '_get_or_create_delivery_pickup' This method should create the pickup if it doesn't exist and return the id in both cases.
The js should call this method when the user select a pickup and then call the [cart/set_delivery_pickup](https://github.com/shopinvader/odoo-shopinvader-carrier/pull/1/files#diff-ea91e78f2a386168e3c5975bcf7fe791R38) with the result. No override of the cart service here.